src/main/scala/mesosphere/marathon/state/ResourceRole.scala

```diff
@@ -1,5 +1,38 @@
 package mesosphere.marathon.state
 
+import com.wix.accord._
+import com.wix.accord.dsl._
+import mesosphere.marathon.api.v2.Validation._
+
 object ResourceRole {
   val Unreserved = "*"
+
+  // NOTE: The validators below use conjunction over `isTrue` in order to provide more user-friendly error messages.
+  //       For example, `char is notEqualTo('\u0020')` would print something like "char is equal to  " as opposed to
+  //       "A role name must not include a space (\x20) character".
+
+  val validResourceRole: Validator[String] = {
+    val message = "A role name must not be %s."
+    isTrue[String](message.format("\"\"")) { role => role != ""; } and
+      isTrue[String](message.format("\".\"")) { role => role != "."; } and
+      isTrue[String](message.format("\"..\"")) { role => role != ".."; } and
+      isTrue[String]("A role name must not start with a '-'.") { role => !role.startsWith("-") } and
+      validator[String] { role =>
+        role.each is valid(validResourceRoleChar)
+      }
+  }
+
+  val validResourceRoleChar: Validator[Char] = {
```

Comment:
Looks like you are a big fan of `isTrue` - which is ok.
Just want to show, that you can also define the validator like this:
```
  val validResourceRoleChar = validator[Char] { char =>
    char is notEqualTo('\u0009')
    char is notEqualTo('xxx`)
    char is notEqualTo('yyy`)
  }
```